pytket/qir/conversion/conversion.py

```diff
@@ -0,0 +1,724 @@
+                pyqir.FunctionType(
+                    pyqir.Type.void(self.module.module.context),
+                    [pyqir.qubit_type(self.module.module.context)] * index,
+                ),
+            )
+
+        module.builder.call(
+            self.barrier[index],  # type: ignore
+            [*qir_qubits],
+        )
+
+    def _add_group_op(
+        self, module: tketqirModule, index: int, qir_qubits: Sequence
+    ) -> None:
+        # __quantum__qis__group1__body()
+        if self.group[index] == None:
+            self.group[index] = self.module.module.add_external_function(
+                f"__quantum__qis__group{index}__body",
+                pyqir.FunctionType(
+                    pyqir.Type.void(self.module.module.context),
+                    [pyqir.qubit_type(self.module.module.context)] * index,
+                ),
+            )
+
+        module.builder.call(
+            self.group[index],  # type: ignore
+            [*qir_qubits],
+        )
+
+    def _add_order_op(
+        self, module: tketqirModule, index: int, qir_qubits: Sequence
+    ) -> None:
+        # __quantum__qis__order1__body()
+        if self.order[index] == None:
+            self.order[index] = self.module.module.add_external_function(
+                f"__quantum__qis__order{index}__body",
+                pyqir.FunctionType(
+                    pyqir.Type.void(self.module.module.context),
+                    [pyqir.qubit_type(self.module.module.context)] * index,
+                ),
+            )
+
+        module.builder.call(
+            self.order[index],  # type: ignore
+            [*qir_qubits],
+        )
+
+    def _add_sleep_op(
+        self, module: tketqirModule, index: int, qir_qubits: Sequence, duration: float
+    ) -> None:
+        # __quantum__qis__sleep1__body()
+        if self.sleep[index] == None:
+            paramlist = [pyqir.qubit_type(self.module.module.context)] * index
+            paramlist.append(
+                pyqir.Type.double(self.module.module.context)
+            )  # add float parameter
+            self.sleep[index] = self.module.module.add_external_function(
+                f"__quantum__qis__sleep{index}__body",
+                pyqir.FunctionType(
+                    pyqir.Type.void(self.module.module.context),
+                    paramlist,
+                ),
+            )
+
+        module.builder.call(
+            self.sleep[index],  # type: ignore
+            [
+                *qir_qubits,
+                pyqir.const(pyqir.Type.double(self.module.module.context), duration),
+            ],
+        )
+
+    def _rebase_command_to_gateset(self, command: Command) -> Optional[Circuit]:
+        """Rebase to the target gateset if needed."""
+        optype = command.op.type
+        params = command.op.params
+        args = command.args
+        if optype not in self.module.gateset.base_gateset:
+            circuit = Circuit(self.circuit.n_qubits, self.circuit.n_bits)
```

Comment:
This creates a `Circuit` with default quantum and classical registers. If the `Command` has qubits or bits from non-default registers in its `args` then `add_gate()` will fail.

Reply:
As discussed this should not be an issue with the type of circuit that is allowed here.